driver/src/main/java/org/neo4j/driver/v1/Config.java

```diff
@@ -241,32 +241,44 @@ public Config toConfig()
     }
 
     /**
-     * A configuration to configure TLS authentication
+     * Control the level of encryption to require
+     */
+    public enum EncryptionLevel
+    {
+        /** With this level, the driver will only connect to the server if it can do it without encryption. */
+        REJECTED,
```

Comment:
It's certainly possible - the driver would try encrypted first, and degrade to non-encrypted. `OPTIONAL` was part of the design and was originally in the DIP, but we agreed to wait to include it due to time constraints. Since this is a useful operational feature, and the overhead of leaving space here (calling this `EncryptionPolicy.REQUIRED/REJECTED` vs `EncryptionPolicy.SECURE/INSECURE` as you suggest) is minimal, I think the alternative paints us into a corner for very little benefit while this leaves a nice door open to add this feature later.